dbt/parsely_dpl_redshift/README.md

```diff
@@ -0,0 +1,43 @@
+### Redshift dbt models for Parse.ly DPL
+
+---
+## Parse.ly DPL DBT
+A "data-build tool" for your Parse.ly data that automates SQL table creation for your data pipeline. It creates queryable tables for things like pageviews, sessions, loyalty users, subscribers, engagement levels, read-time, and handles the incremental loading of new data from S3 to your SQL tables. By handling this setup work for you, the data-build tool reduces configuration time and lets you get started writing your own custom queries more quickly.
+
+## Schemas/models
+- *Users*
+  Table Grain: 1 row per unique user ID defined by IP address and cookie. The table that provides the Parse.ly DPL lifetime engagement of a user including defining the user as Loyalty and Rolling 30 day loyalty.
+- *Sessions*
+  Table Grain: 1 row per user sessions. A session is defined as any activity by one user without being idle for more than 30 minutes. The session table includes total engagement and pageview metrics for the entire session as well as the user-types at the time of the session. This allows for simplified identification of conversions into loyalty users and subscribers.
+- *Content*
+  Table Grain: 1 row per article or video. This table contains only the most recent metadata for each article or video and aids in simplified reporting and aggregation when metadata is changed throughout the lifetime of an article.
+- *Campaigns*
+  Table Grain: 1 row per campaign. This table contains only the most recent description for each campaign.
+- *Pageviews*
+  Table Grain: 1 row per pageview. This table contains the referrer, campaign, timestamps, engaged time, and at-time-of-engagement metadata info for each pageview. The pageviews are organized to view the order and flow of pageviews within a session for one user.
+- *Videoviews*
+  Table Grain: 1 row per videoview. This table contains the referrer, campaign, timestamps, engaged time, and at-time-of-engagement metadata info for each video view. The videviews are organized to view the order and flow of videoviews within a session for one user.
+- *Custom events*
+  Table Grain: 1 row per custom event sent through the Parse.ly DPL. This is any event that is not: pageview, heartbeat, videostart, or vheartbeat. These can be specified in the dbt_project.yml file and contain keys to join to users, sessions, content, and campaigns.
+
+
+## How to get started
+- Install DBT
+```pip install -r requirements.txt```
```

Comment:
After thinking about this a little more, let's make the DBT subproject a console script, like these: https://github.com/Parsely/parsely_raw_data/blob/master/setup.py#L87-L95

This setup would allow a user to do the following:
 - pip install `parsely_raw_data` into a virtualenv somehwere
 - run `parsely_dbt redshift <args>` from anywhere

The current setup requires a user to have parsely raw data installed in an active virtualenv, as well as work inside the directory for the project itself. Making a console script adds your project in a similar place that people can already do other tasks, and not need to worry about where to find the code on their system.

This will take a little bit more work, but I think it will be worthwhile.
 - change imports in dbt/parsely_dpl_redshift/redshift.py` to be relative imports
 - add a generic script at DBT root level that allows the user to pass a datasource (i.e. `redshift`) and then selects the appropriate datasource script from there
 - add a console script (like above) that calls `dbt/<script name>` or whatever

This adds a little indirection in the form of the `dbt/scripts`, but that allows you to add new datasources as you go, i.e. `parsely_dbt bigquery <args>...` etc., without having to add a new console script altogether. (Otherwise you'd need to add a separate console script for each new datasource -- please contradict me if you think it's better to have a separate one for each, though, like if the args would be very very different.)

Reply:
Added: 'parsely_dbt_redshift = parsely_raw_data/dbt/parsely_dpl_redshift.redshift:main'
to setup.py

I think I may have set this up differently than you recommended. As it is now, it will not require a generic script at the DBT root level as the root level setup.py entry_point is defined here for the data source (as is for the parsely_raw_data redshift and bigquery distinction as well)

Reply:
:+1: